### ISSUE TITLE
[move-package][enhancement] better error message for `move package test`

### ISSUE THREAD

**Issue by sblackshear** [diem/diem#9440]
Running `move package test` in a directory without a Move.toml file yields "Error: No such file or directory (os error 2)", which is hard to diagnose if (e.g.) a developer is trying to run tests from inside the `tests` directory of a package. 

A more helpful error message that explains this and/or checks one level up in the directory hierarchy for a Move.toml could be useful.

To repro:

```
cd diem/language/move-stdlib/tests
move package test
```

**Comment by tzakian** [comment]
This is another one that should be fixed/mitigated with the changes in #9491. Now if you run a Move package command from anywhere in the package it should work.

However, I don't think we give a nice message if you try running it outside of the package -- probably worth adding, so I won't close this issue just yet.

**Comment by tzakian** [comment]
Going to close this issue out -- we now give back a nice message if you try running a package command outside of a package, e.g.,
```
$ mpm test # Not in a package dir
Error: Unable to find package manifest in 'some_dir' or in its parents
```

Feel free to reopen if this doesn't address the issue -- happy to discuss some more and revisit it
